### PR TITLE
[#2808] fixing spatial coverage radio button issue

### PR DESCRIPTION
**Pull request by sblack-usu**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. add a file aggregation with special coverage.  Validate the coverage is updated at the resource level on upload.  Go into edit mode, click the aggregation and look at spatial coverage under Content Metadata.  Validate it is correct.  Update resource level spatial coverage.  Validate spatial coverage on the aggregation was not updated with the resource level coverage.

### CODE REVIEW

**Comment by sblack-usu** [comment]
@pkdash - I'm seeing an error logged referencing a type field that doesn't exist on CoverageTemporalForm.  I'm having trouble finding where this is happening.  Any ideas?
```
[14/Nov/2018 17:12:01] WARNING [root:110] Could not resolve form field 'type'.
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/crispy_forms/utils.py", line 81, in render_field
    bound_field = form[field]
  File "/usr/local/lib/python2.7/site-packages/django/forms/forms.py", line 164, in __getitem__
    ', '.join(sorted(f for f in self.fields)),
KeyError: u"Key 'type' not found in 'CoverageTemporalForm'. Choices are: end, start."
```

**Comment by sblack-usu** [comment]
@pkdash - nvm, I found it and updated this PR.

**Comment by pkdash** [comment]
@sblack-usu Not sure how it was not failing to render the form with 'type' being there. Glad you fixed it.

File: theme/static/js/resourceLandingAjax.js [hunk 1]
```diff
@@ -1591,23 +1593,86 @@ function setFileTypeSpatialCoverageFormFields(logical_type){
...
-        $id_type_filetype_div.find("#id_type_2").attr('onclick', 'return false');
+        $id_type_filetype_div.find(radioBoxSelector).attr('onclick', 'return false');
+        $id_type_filetype_div.find(radioPointSelector).attr('onclick', 'return false');
         if (logical_type !== "RefTimeseriesLogicalFile"){
-            $id_type_filetype_div.find("#id_type_1").attr('checked', 'checked');
+            $id_type_filetype_div.find(radioBoxSelector).attr('checked', 'checked');
+        }
+        $id_type_filetype_div.find(radioPointSelector).attr('disabled', true);
+        $id_type_filetype_div.find(radioPointSelector).parent().closest("label").addClass("text-muted");
+        if (logical_type === "NetCDFLogicalFile" || logical_type === "GeoRasterLogicalFile"){
```
Comment (pkdash) [inline theme/static/js/resourceLandingAjax.js]: This 'if' block code is not needed for this fix. This is part of my file set branch work where I am adding the feature to edit spatial coverage using the map interface.

File: theme/static/js/resourceLandingAjax.js [hunk 2]
```diff
@@ -1591,23 +1593,86 @@ function setFileTypeSpatialCoverageFormFields(logical_type){
...
         }
-        $id_type_filetype_div.find("#id_type_2").attr('disabled', true);
-        $id_type_filetype_div.find("#id_type_2").parent().closest("label").addClass("text-muted");
     }
     else {
-        // file type is "GenericLogicalFile" - allow changing coverage type
-        $id_type_filetype_div.find("input:radio").change(updateEditCoverageState);
+        // file type is "GenericLogicalFile" or "FileSetLogicalFile" - allow changing coverage type
+        $id_type_filetype_div.find("input:radio").change(updateEditCoverageStateFileType);
+        var onSpatialCoverageDelete = function () {
```
Comment (pkdash) [inline theme/static/js/resourceLandingAjax.js]: Again this function is not needed for this fix - it is specific to file set work.

File: theme/static/js/resourceLandingAjax.js [hunk 3]
```diff
@@ -1591,23 +1593,86 @@ function setFileTypeSpatialCoverageFormFields(logical_type){
...
+            var url = formSpatialCoverage.attr('action');
+            url = url.replace('update-file-metadata', 'delete-file-coverage');
+            $btnDeleteSpatialCoverage.unbind('click');
+            $btnDeleteSpatialCoverage.click(function () {
+                deleteFileTypeSpatialCoverage(url, $btnDeleteSpatialCoverage);
+
+            })
+        };
+
+        var addSpatialCoverageLink = function () {
```
Comment (pkdash) [inline theme/static/js/resourceLandingAjax.js]: Same comment as above.

File: theme/static/js/resourceLandingAjax.js [hunk 4]
```diff
@@ -1591,23 +1593,86 @@ function setFileTypeSpatialCoverageFormFields(logical_type){
...
+            var $spatialForm = $("#id-coverage-spatial-filetype");
+            var deleteLink = '<a id="id-btn-delete-spatial-filetype" type="button" style="display: block;" class="pull-right"><span class="glyphicon glyphicon-trash icon-button btn-remove"></span>';
+            $spatialForm.find('legend').html('Spatial Coverage' + deleteLink);
+            $btnDeleteSpatialCoverage = $("#id-btn-delete-spatial-filetype");
+            return $btnDeleteSpatialCoverage;
+        };
+        // set spatial form attribute 'data-coordinates-type' to point or rectangle
+        if ($id_type_filetype_div.find(radioBoxSelector).attr("checked") === "checked"){
```
Comment (pkdash) [inline theme/static/js/resourceLandingAjax.js]: Same comment for this 'if' block.

File: theme/static/js/resourceLandingAjax.js [hunk 5]
```diff
@@ -1591,23 +1593,86 @@ function setFileTypeSpatialCoverageFormFields(logical_type){
...
+        if(bindCoordinatesPicker){
+            $("#id-coverage-spatial-filetype").coordinatesPicker();
+        }
     }
 
     // #id_type_1 is the box radio button
-    if ($id_type_filetype_div.find("#id_type_1").attr("checked") == "checked" ||
-        (logical_type != 'GeoFeatureLogicalFile' && logical_type != 'RefTimeseriesLogicalFile' && logical_type != 'GenericLogicalFile')) {
-        // coverage type is box
+    if ($id_type_filetype_div.find(radioBoxSelector).attr("checked") === "checked" ||
```
Comment (pkdash) [inline theme/static/js/resourceLandingAjax.js]: This only change to the if statement we need is "#id_type_1" -> radioBoxSelector rest of the if statement should not be changed.

File: theme/static/js/resourceLandingAjax.js [hunk 6]
```diff
@@ -1685,7 +1750,42 @@ function updateResourceSpatialCoverage(spatialCoverage) {
     }
 }
 
-// updates the UI temporal coverage elements
+function addFileTypeTemporalCoverageDeleteLink() {
```
Comment (pkdash) [inline theme/static/js/resourceLandingAjax.js]: Again this is not needed for this fix.

File: theme/static/js/resourceLandingAjax.js [hunk 7]
```diff
@@ -1685,7 +1750,42 @@ function updateResourceSpatialCoverage(spatialCoverage) {
...
+    var $temporalForm = $("#id-coverage-temporal-file-type");
+    var deleteLink = '<a id="id-btn-delete-temporal-filetype" type="button" style="display: block;" class="pull-right"><span class="glyphicon glyphicon-trash icon-button btn-remove"></span>';
+    $temporalForm.find('legend').html('Temporal Coverage' + deleteLink);
+    var $btnDeleteTemporalCoverage = $("#id-btn-delete-temporal-filetype");
+    return $btnDeleteTemporalCoverage;
+}
+
+function setFileTypeTemporalCoverageDeleteOption(logicalFileType) {
```
Comment (pkdash) [inline theme/static/js/resourceLandingAjax.js]: Not needed for this fix.

File: theme/static/js/resourceLandingAjax.js [hunk 8]
```diff
@@ -1685,7 +1750,42 @@ function updateResourceSpatialCoverage(spatialCoverage) {
...
+    // show/hide delete option for temporal coverage at aggregation level
+    var $btnDeleteTemporalCoverage = $("#id-btn-delete-temporal-filetype");
+    if(!$btnDeleteTemporalCoverage.length) {
+        // delete option doesn't exist - so add it
+        $btnDeleteTemporalCoverage = addFileTypeTemporalCoverageDeleteLink();
+    }
+
+    var $formTemporalCoverage = $btnDeleteTemporalCoverage.closest('form');
+    if (logicalFileType === 'GenericLogicalFile' || logicalFileType === 'FileSetLogicalFile') {
```
Comment (pkdash) [inline theme/static/js/resourceLandingAjax.js]: Not needed for this fix.

File: theme/static/js/resourceLandingAjax.js [hunk 9]
```diff
@@ -1695,6 +1795,70 @@ function updateResourceTemporalCoverage(temporalCoverage) {
     initializeDatePickers();
 }
 
+// updates the UI spatial coverage elements for the aggregation
+function updateAggregationSpatialCoverageUI(spatialCoverage, logicalFileID, elementID) {
```
Comment (pkdash) [inline theme/static/js/resourceLandingAjax.js]: This function is not needed for this fix.

File: theme/static/js/resourceLandingAjax.js [hunk 10]
```diff
@@ -1695,6 +1795,70 @@ function updateResourceTemporalCoverage(temporalCoverage) {
...
+        $("#id_southlimit_filetype").parent().closest("#div_id_southlimit").show();
+        $("#id_westlimit_filetype").parent().closest("#div_id_westlimit").show();
+        $("#id_north_filetype").parent().closest("#div_id_north").hide();
+        $("#id_east_filetype").parent().closest("#div_id_east").hide();
+    }
+    initMapFileType();
+}
+
+// updates the UI temporal coverage elements for the aggregation
+function updateAggregationTemporalCoverage(temporalCoverage, logicalFileID, coverageElementID) {
```
Comment (pkdash) [inline theme/static/js/resourceLandingAjax.js]: This function is not needed for this fix.

**Comment by sblack-usu** [comment]
@pkdash - I decide to amend the cherry-picked commit with the changes you described here instead of adding another commit.  I hope this makes it easier to merge your ticket when it is ready.

New PR here:
https://github.com/hydroshare/hydroshare/pull/3010

**Pull request closed without merging**